pubsub/aws/snssqs/snssqs.go

```diff
@@ -601,12 +606,13 @@ func (s *snsSqs) consumeSubscription(ctx context.Context, queueInfo, deadLetters
 		// iteration. Therefore, a global backoff (to the internal backoff) is used (sqsPullExponentialBackoff).
 		messageResponse, err := s.sqsClient.ReceiveMessageWithContext(ctx, receiveMessageInput)
 		if err != nil {
-			if err == context.Canceled || err == context.DeadlineExceeded {
+			if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) || ctx.Err() != nil {
```

Comment:
```suggestion
			if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
```

Reply:
why? the context might be cancelled while we pull messages but still haven't started a new for loop iteration (that verifies the ctx isn't cancelled, but just at the top of the loop)